tests/block_builder_test.rs

```diff
@@ -45,13 +45,8 @@ async fn test_handle_build() {
     let ru_provider = RootProvider::<Ethereum>::new_http(anvil_instance.endpoint_url());
     let host_provider = config.connect_host_provider().await.unwrap();
 
-    let block_env = EnvTask::new(
-        config.clone(),
-        config.connect_host_provider().await.unwrap(),
-        ru_provider.clone(),
-    )
-    .spawn()
-    .0;
+    let block_env =
+        EnvTask::new(config.clone(), host_provider.clone(), ru_provider.clone()).spawn().0;
```

Comment:
this is a drive-by test fix to avoid connecting provider twice which wasn't necessary